Synthesio/synthesio_api.py

```diff
@@ -1,147 +1,209 @@
+            if '' in or_new_keywords:
+                print('illegal query! Please double check!')
+                return None
+            else:
+                sub_query_string = ""
+                for i in range(len(or_new_keywords)):
+                    if i== 0:
+                        sub_query_string += "(\"" + or_new_keywords[i] + "\" " + or_boolean[i].upper() + " "
+                    elif i== len(or_new_keywords)-1:
+                        sub_query_string += "\"" + or_new_keywords[i] + "\")"
+                    else:
+                        sub_query_string += "\"" + or_new_keywords[i]  + "\" " + or_boolean[i].upper() + " "
+
+            new_keywords.append(sub_query_string)
+        
+        # if "or" boolean not exist
+        # just strip the trailing spaces and wrap them into " "
+        else:
+            new_keywords.append("\"" +  " ".join(k.split()) + "\"")
+                 
+    if '\"\"' in new_keywords:
+        print('illegal query! Please double check!')
+        return None
+    
+    else:
+        
+        new_query_string = ""
+        for i in range(len(new_keywords)):
+            if i == len(new_keywords)-1:
+                new_query_string += new_keywords[i] 
+            else:
+                new_query_string += new_keywords[i]  + " " + boolean[i].upper() + " "
+        
+        return new_query_string
+
+
+    
+        
+def get_mentions_query_single(words, research_report_id, size, access_token):
+    formatted_query = format_query(words)
+    payload = {'filters': {'query': formatted_query}}
+    mentions_query_single = post_requests_data_json('mention/v2/reports/' + research_report_id + '/_search?size=' + size + '&sort=desc', headers_json, payload, access_token)
+    mentions_query_single_data = mentions_query_single['data']
+    return mentions_query_single_data
+
+def normalize(check_query):
+    i = 0 
+    while i < len(check_query):
+        if check_query[i].keys() != master_dict.keys():
+            check_query[i].update(dict.fromkeys(set(master_dict).difference(check_query[i]), 'None'))
+            i+=1
+
+def get_values(keys_values): 
+    values = []
+    i = 0
+    while i < len(keys_values):
+        ordered_dic_keys_values = OrderedDict(sorted(keys_values[i].items()))
+        str_values_in_dic = [str(value) for key, value in ordered_dic_keys_values.items()]
+        replaced_values = [item.replace(',',' ') for item in str_values_in_dic]
+        values.append(replaced_values)
+        i+=1
+    return values
+        
+def print_file(title, content):
+    csv_title = title
+    csv_content = content
+    with open(title + '.csv', 'w',newline="") as makecsvfile:
+        labels = [key for key in ordered_master_dict.keys()]
+        data = csv.writer(makecsvfile, delimiter=',',
+                          quotechar='|', quoting=csv.QUOTE_MINIMAL)
+        data.writerow(labels)
+        for row in get_values(csv_content):
+            try:
+                [data.writerow(row) ]

```

Comment:
ah here should be try: data.writerow(row) without the []... I forget. But the output seems just fine to me though